src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -0,0 +1,332 @@
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import co.cask.gcp.common.GCPConfig;
+import co.cask.gcp.common.GCPUtils;
+import com.google.cloud.speech.v1.RecognitionAudio;
+import com.google.cloud.speech.v1.RecognitionConfig;
+import com.google.cloud.speech.v1.RecognizeResponse;
+import com.google.cloud.speech.v1.SpeechClient;
+import com.google.cloud.speech.v1.SpeechRecognitionAlternative;
+import com.google.cloud.speech.v1.SpeechRecognitionResult;
+import com.google.cloud.speech.v1.SpeechSettings;
+import com.google.protobuf.ByteString;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+import javax.ws.rs.Path;
+
+/**
+ * Google Cloud Speech-To-Text Transform which transcripts audio
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name(SpeechToTextTransform.NAME)
+@Description(SpeechToTextTransform.DESCRIPTION)
+public class SpeechToTextTransform extends Transform<StructuredRecord, StructuredRecord> {
+  public static final String NAME = "SpeechToText";
+  public static final String DESCRIPTION = "Converts audio files to text by applying powerful neural network models.";
+  private SpeechTransformConfig config;
+  private Schema schema;
+  private RecognitionConfig recognitionConfig;
+  private SpeechClient speech;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgumentException {
+    super.configurePipeline(configurer);
+    Schema inputSchema = configurer.getStageConfigurer().getInputSchema();
+    // If a inputSchema and audioField is available then verify that the schema does contain the given audioField
+    // and that the type is byte array. This will allow to fail fast i.e. during deployment time.
+    if (inputSchema != null && !config.containsMacro("audiofield")) {
+      if (inputSchema.getField(config.audioField) == null) {
+        throw new IllegalArgumentException(String.format("Field %s must exists.", config.audioField));
+      }
+      Schema.Field field = inputSchema.getField(config.audioField);
+      Schema fieldSchema = field.getSchema();
+      Schema.Type fieldType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
+      if (fieldType != Schema.Type.BYTES) {
+        throw new IllegalArgumentException(String.format("Field '%s' must be of type 'bytes' but is of type " +
+                                                           "'%s'.", field.getName(), fieldType));
+      }
+    }
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    schema = config.getSchema();
```

Comment:
This transform should not use a user-supplied schema. That opens the door to all sorts of weirdness. What if they change the type of an input field?

Instead, the schema here should be the output schema, assuming the input schema is not null.